dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -473,6 +475,18 @@ void ExchangeReceiverBase<RPCContext>::cancel()
 {
     setEndState(ExchangeReceiverState::CANCELED);
     cancelAllMsgChannels();
+    if (is_receiver_for_tiflash_storage)
```

Comment:
`ExchangeReceiverBase<RPCContext>::cancel()` may be called multiple times.
Better to check ExchangeReceiver::status first.

Reply:
Done